test/jdk/javax/security/auth/callback/PasswordCleanup.java

```diff
@@ -0,0 +1,87 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8284910
+ * @summary Buffer clean in PasswordCallback
+ */
+
+import javax.security.auth.callback.PasswordCallback;
+import java.util.WeakHashMap;
+
+public final class PasswordCleanup {
+    private final static WeakHashMap<PasswordCallback, ?> weakHashMap =
+            new WeakHashMap<>();
+
+    public static void main(String[] args) throws Exception {
+        // Test password clearing at finalization.
+        clearAtCollection();
+
+        // Test password clearing with the specific method.
+        clearWithMethod();
+    }
+
+    private static void clearAtCollection() throws Exception {
+        // Create an object
+        PasswordCallback passwordCallback =
+                new PasswordCallback("Password: ", false);
+        passwordCallback.setPassword("ThisIsAPassword".toCharArray());
+        passwordCallback.clearPassword();
+        weakHashMap.put(passwordCallback, null);
+        passwordCallback = null;
+
+        // Check the clearing
+        checkClearing();
+    }
+
+    private static void clearWithMethod() throws Exception {
+        // Create an object
+        PasswordCallback passwordCallback =
+                new PasswordCallback("Password: ", false);
+        passwordCallback.setPassword("ThisIsAPassword".toCharArray());
+
+        // Use password clear method.
+        passwordCallback.clearPassword();
+
+        weakHashMap.put(passwordCallback, null);
+        passwordCallback = null;
+
+        // Check the clearing
+        checkClearing();
+    }
+
+    private static void checkClearing() throws Exception {
+        // Wait to trigger the cleanup.
+        for (int i = 0; i < 10 && weakHashMap.size() != 0; i++) {
+            System.gc();
+            Thread.sleep(100);
+        }
+
+        // Check if the object has been collected.
+        if (weakHashMap.size() > 0) {
+            throw new RuntimeException("GSSName object is not released");
```

Comment:
Did you mean to say "PasswordCallback object is not released"?

Reply:
Ooops, bad copy and past of mine.